limitations: Added read buffer size information. (#141)
bisdn/bisdn.github.io: limitations_and_known_issues.md

# Open issues

> ## Socket receive buffer size cause baseboxd to miss Netlink events.

Reviewer: ```suggestion
## Socket receive buffer size causes baseboxd to miss netlink events
```

Affected versions: 3.0 - current

> The maximum receive buffer size ``net.core.rmem_max`` is lower than the specifically set default buffer size ``net.core.rmem_default``. Because baseboxd creates its Netlink read buffer based on the max value, Netlink events can be missed if a large burst of events happens. This can cause the ASIC state to be out of synch.

Reviewer: ```suggestion
BISDN Linux increases `net.core.rmem_default` while leaving `net.core.rmem_max` at the default value which is lower than the new `net.core.rmem_default`. Because baseboxd creates its netlink read buffer based on the max value, a large burst of netlink events may result in netlink messages being lost, with baseboxd failing to fully synchronize the ASIC state with the kernel state.
```

Reviewer: I don't know what the double backticks are supposed to do in github markdown, so I replaced them with single backticks.

> A simple workaround for this is to add the line ``net.core.rmem_max=8388608`` to ``/etc/sysctl.d/20-network-io.conf``

Reviewer: ```suggestion
The solution is to add the line `net.core.rmem_max=8388608` to `/etc/sysctl.d/20-network-io.conf`.
```

Reviewer: As far as issue described in this section is concerned, this should be the solution and not just a workaround.
